### PR TITLE
chore: bump rust toolchain from 1.41.0 to 1.45.2

### PR DESCRIPTION
**Pull request by yangby-cryptape**
- Fix clippy warings, except [`clippy::mutable_key_type`].
- Allow [`clippy::mutable_key_type`].

[`clippy::mutable_key_type`]: https://rust-lang.github.io/rust-clippy/master/index.html#mutable_key_type

### CODE REVIEW

File: rpc/src/error.rs [hunk 1]
```diff
@@ -73,15 +73,19 @@ impl RPCError {
     }
 
     pub fn from_ckb_error(err: CKBError) -> Error {
-        use ckb_error::ErrorKind::*;
+        use ckb_error::ErrorKind;
```
Comment (quake) [inline rpc/src/error.rs:76]: move this line to file header section

Comment (yangby-cryptape) [reply rpc/src/error.rs]: DONE!

**Comment by zhangsoledad** [comment]
https://github.com/rust-lang/rust/issues/74739

**Comment by doitian** [comment]
Better upgrade to 1.45.1

**Comment by driftluo** [comment]
wait https://github.com/rust-lang/rust/pull/74958

**Comment by doitian** [comment]
CKB Docker Builder PR-3 unblocked

**Comment by yangby-cryptape** [comment]
bors r=quake,driftluo

**Comment by doitian** [comment]
Approved. The target release is v0.36.0

**Comment by yangby-cryptape** [comment]
bors r=quake,driftluo,doitian

**Pull request merged**